test/src/unit-cppapi-group.cc

```diff
@@ -1102,38 +1102,219 @@ TEST_CASE(
 }
 
 TEST_CASE(
-    "C++ API: Group with Relative URI members, write/read, rest",
-    "[cppapi][group][relative][rest]") {
+    "C++ API: Group with relative URI members, write/read, rest",
+    "[cppapi][group][create][relative][rest]") {
   VFSTestSetup vfs_test_setup;
+  if (!vfs_test_setup.is_rest()) {
+    // 3.0 REST must add the child asset as a group member for this to pass.
+    // Legacy REST must return an error for this to pass.
+    SKIP("This test relies on serverside REST logic");
+  }
   tiledb::Context ctx{vfs_test_setup.ctx()};
-  auto group_name{vfs_test_setup.array_uri("groups_relative")};
-  auto subgroup_name = group_name + "/subgroup";
+  auto group_uri{vfs_test_setup.array_uri("groups_relative")};
+  tiledb::create_group(ctx, group_uri);
+
+  auto member_uri_str = group_uri + "/relative_group";
+  if (vfs_test_setup.is_legacy_rest()) {
+    // Legacy REST does not support relative group members.
+    auto group = tiledb::Group(ctx, group_uri, TILEDB_WRITE);
+    tiledb::create_group(ctx, member_uri_str);
+    CHECK_NOTHROW(group.add_member("relative_group", true, "subgroup"));
+    CHECK_THROWS_WITH(
+        group.close(),
+        Catch::Matchers::ContainsSubstring(
+            "relative paths are not yet supported for cloud groups"));
+  } else {
+    // For 3.0 REST construct an asset path relative to the parent path above.
+    // 3.0 REST will create a child asset under the parent group's asset path
+    // _and_ register the asset as a member of `groups_relative`.
+    tiledb::sm::URI::RESTURIComponents components;
+    SECTION("Creating a group member as a child asset") {
+      tiledb::sm::URI member_uri(member_uri_str);
+      CHECK(
+          member_uri
+              .get_rest_components(vfs_test_setup.is_legacy_rest(), &components)
+              .ok());
+      // Append the child group to the server asset path in the URI.
+      components.server_path.resize(components.server_path.find('/'));
+      components.server_path += "/groups_relative";
+      member_uri_str = "tiledb://" + components.server_namespace + "/" +
+                       components.server_path + "/" + components.asset_storage;
+      tiledb::create_group(ctx, member_uri_str);
+
+      auto group = tiledb::Group(ctx, group_uri, TILEDB_READ);
+      auto member = group.member("relative_group");
+      CHECK(member.type() == tiledb::Object::Type::Group);
+      CHECK(member.name() == "relative_group");
+      std::string expected_uri =
+          build_tiledb_uri(member_uri, "groups_relative/relative_group");
+      CHECK(member.uri() == expected_uri);
+      CHECK(group.is_relative("relative_group"));
```

Comment:
Your Server PR made this code produce a relative member instead of an absolute? AFAIK the behavior that creating a group/array inside a group results in adding it as a member existed.

Reply:
Vlasis made a similar comment on my server side PR so I cc'd you in my reply there for some more detail. But the TLDR is yes, currently creating a group member using an asset path with a parent asset that is an existing group on REST will build a relative storage URI server side and create the new array / group member relative to its parent group on backend storage. I found that this is the case for both array and group creation routes today prior to any group member relative URI changes.